### PR TITLE
[BEAM-6620, BEAM-3606] Stop shading by default.

### PR DESCRIPTION
**Pull request by lukecwik**
This stops shading guava by default in all modules that don't have a custom shading configuration which is all modules except for:
* model/*
* sdks/java/core
* sdks/java/extensions/kryo
* sdks/java/extensions/sql
* sdks/java/extensions/sql/jdbc
* sdks/java/harness
* runners/spark/job-server
* runners/direct-java
* runners/samza/job-server
* runners/google-cloud-dataflow-java/worker
* runners/google-cloud-dataflow-java/worker/legacy-worker
* runners/google-cloud-dataflow-java/worker/windmill
* vendor/*

This migrates all modules except for the ones listed above so that:
* publishing is based off of the unshaded jar and the unshaded test jar.
* all intra module dependences use the compile (default) or testCompile configuration.
* all dependencies within the module to use compile/testCompile/testRuntimeClasspath instead of shadow/shadowTest/shadowTestRuntimeClasspath

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by lukecwik** [comment]
Run Java PostCommit

**Comment by lukecwik** [comment]
Run Data flow PostCommit

**Comment by lukecwik** [comment]
Run Spark PostCommit

**Comment by lukecwik** [comment]
Run Flink PostCommit

**Comment by lukecwik** [comment]
Run Flink ValidatesRunner

**Comment by lukecwik** [comment]
Run Dataflow ValidatesRunner

**Comment by lukecwik** [comment]
Run Spark ValidatesRunner

**Comment by iemejia** [comment]
Run Spark ValidatesRunner

**Comment by iemejia** [comment]
Oups restarting it by mistake actually it seems there are some issues of the `NoSuchMethodError` and `NoClassDefFoundError` kind with the Spark changes. [previus run link](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_PR/90/#showFailuresLink)

**Comment by lukecwik** [comment]
I figured out that the Spark ValidatesRunner tests were only working because they were bringing in hadoop-format in a way where certain parts of cassandra weren't being brought in. This avoided an issue where Cassandra would have brought in net.jpountz.lz4:lz4 which is incompatible with the lz4-java that Spark depends on.

**Comment by lukecwik** [comment]
Run Flink ValidatesRunner

**Comment by lukecwik** [comment]
Run Dataflow ValidatesRunner

**Comment by lukecwik** [comment]
Run Java PostCommit

**Comment by lukecwik** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by lukecwik** [comment]
Run Flink ValidatesRunner

**Comment by lukecwik** [comment]
Run Spark ValidatesRunner

**Comment by lukecwik** [comment]
I also was able to publish a release into the [staging repository](https://repository.apache.org/content/groups/snapshots/org/apache/beam/) and compared the newly generated poms artifact-2.14.0-20190605.*-30.pom against the previously nightly snapshot of artifact-2.14.0-20190605.*-28.pom for the following projects as a spot check and found no differences in those poms:
beam-sdks-java-core
beam-sdks-java-fn-execution
beam-runners-spark

**Comment by lukecwik** [comment]
R: @apilloud @adude3141

**Comment by lukecwik** [comment]
Run Java PostCommit

**Comment by lukecwik** [comment]
Run Dataflow ValidatesRunner

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 1]
```diff
@@ -558,17 +561,8 @@ class BeamModulePlugin implements Plugin<Project> {
     // https://github.com/johnrengelman/shadow/blob/98191096a94674245c7b3e63975df9e14f67074e/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/internal/DefaultDependencyFilter.groovy#L123
     project.ext.DEFAULT_SHADOW_CLOSURE = {
       dependencies {
-        include(dependency(project.library.java.guava))
-      }
-      // guava uses the com.google.common and com.google.thirdparty package namespaces
-      relocate("com.google.common", project.getJavaRelocatedPath("com.google.common")) {
```
Comment (apilloud) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: This is awesome!

**Comment by ibzib** [comment]
Run Java Spark PortableValidatesRunner Batch

**Pull request merged**